src/features/transfer/TransferTokenForm.tsx

```diff
@@ -111,14 +112,37 @@ function SwapChainsButton({ disabled }: { disabled?: boolean }) {
 function ChainSelectSection({ isReview }: { isReview: boolean }) {
   const warpCore = useWarpCore();
   const chains = useMemo(() => warpCore.getTokenChains(), [warpCore]);
+  const chainMetadata = useChainMetadataMap(chains);
+
+  const { values } = useFormikContext<TransferFormValues>();
+
+  const originRoutes = useMemo(() => {
+    return getCustomListItemField(warpCore, values.origin, chainMetadata);
+  }, [values.origin, warpCore, chainMetadata]);
+
+  const destinationRoutes = useMemo(() => {
+    return getCustomListItemField(warpCore, values.destination, chainMetadata);
+  }, [values.destination, warpCore, chainMetadata]);
```

Comment:
Can we move this into `ChainSelectField`?

Reply:
Reason I didn't do that it is because I think it looks better to separate the logic here instead of having it inside the `ChainSelectField` by doing something like `name === "origin" ? values.destination : values.origin`